crates/std_detect/src/detect/arch/riscv.rs

```diff
@@ -70,61 +87,33 @@ features! {
-    ///
-    /// This document defined the following extensions:
-    ///
-    /// * Zba: `"zba"`
-    /// * Zbb: `"zbb"`
-    /// * Zbc: `"zbc"`
-    /// * Zbs: `"zbs"`
-    ///
-    /// # RISC-V Cryptography Extensions
-    ///
-    /// These extensions are defined in Volume I, Scalar & Entropy Source
-    /// Instructions:
-    ///
-    /// * Zbkb: `"zbkb"`
-    /// * Zbkc: `"zbkc"`
-    /// * Zbkx: `"zbkx"`
-    /// * Zknd: `"zknd"`
-    /// * Zkne: `"zkne"`
-    /// * Zknh: `"zknh"`
-    /// * Zksed: `"zksed"`
-    /// * Zksh: `"zksh"`
-    /// * Zkr: `"zkr"`
-    /// * Zkn: `"zkn"`
-    /// * Zks: `"zks"`
-    /// * Zk: `"zk"`
-    /// * Zkt: `"zkt"`
-    ///
     /// [ISA manual]: https://github.com/riscv/riscv-isa-manual/
     #[unstable(feature = "stdarch_riscv_feature_detection", issue = "111192")]
-    @FEATURE: #[unstable(feature = "stdarch_riscv_feature_detection", issue = "111192")] rv32i: "rv32i";
+    @FEATURE: #[stable(feature = "riscv_ratified", since = "1.76.0")] rv32i: "rv32i";
     /// RV32I Base Integer Instruction Set
-    @FEATURE: #[unstable(feature = "stdarch_riscv_feature_detection", issue = "111192")] zifencei: "zifencei";
+    @FEATURE: #[stable(feature = "riscv_ratified", since = "1.76.0")] zifencei: "zifencei";
     /// "Zifencei" Instruction-Fetch Fence
-    @FEATURE: #[unstable(feature = "stdarch_riscv_feature_detection", issue = "111192")] zihintpause: "zihintpause";
+    @FEATURE: #[stable(feature = "riscv_ratified", since = "1.76.0")] zihintpause: "zihintpause";
     /// "Zihintpause" Pause Hint
-    @FEATURE: #[unstable(feature = "stdarch_riscv_feature_detection", issue = "111192")] rv64i: "rv64i";
+    @FEATURE: #[stable(feature = "riscv_ratified", since = "1.76.0")] rv64i: "rv64i";
     /// RV64I Base Integer Instruction Set
-    @FEATURE: #[unstable(feature = "stdarch_riscv_feature_detection", issue = "111192")] m: "m";
+    @FEATURE: #[stable(feature = "riscv_ratified", since = "1.76.0")] m: "m";
     /// "M" Standard Extension for Integer Multiplication and Division
-    @FEATURE: #[unstable(feature = "stdarch_riscv_feature_detection", issue = "111192")] a: "a";
+    @FEATURE: #[stable(feature = "riscv_ratified", since = "1.76.0")] a: "a";
     /// "A" Standard Extension for Atomic Instructions
-    @FEATURE: #[unstable(feature = "stdarch_riscv_feature_detection", issue = "111192")] zicsr: "zicsr";
+    @FEATURE: #[stable(feature = "riscv_ratified", since = "1.76.0")] zicsr: "zicsr";
     /// "Zicsr", Control and Status Register (CSR) Instructions
-    @FEATURE: #[unstable(feature = "stdarch_riscv_feature_detection", issue = "111192")] zicntr: "zicntr";
+    @FEATURE: #[stable(feature = "riscv_ratified", since = "1.76.0")] zicntr: "zicntr";
     /// "Zicntr", Standard Extension for Base Counters and Timers
-    @FEATURE: #[unstable(feature = "stdarch_riscv_feature_detection", issue = "111192")] zihpm: "zihpm";
+    @FEATURE: #[stable(feature = "riscv_ratified", since = "1.76.0")] zihpm: "zihpm";
```

Comment:
Same here: zicsr, zicntr, zihpm, rv32i, rv64i, zifencei, zihintpause will all never be enabled, because it's not exposed in hwcaps. These should not be stabilized as is.

It's not an issue for feature that are already exposed on the compiler side (e.g. zba, zbb) since in that case we just return the static configuration from `cfg(target_feature)`.

Reply:
These are also removed.